crowbar_framework/app/helpers/barclamp/network_helper.rb

```diff
@@ -17,5 +17,13 @@
 
 module Barclamp
   module NetworkHelper
+    def self.wrap_ip(address)
```

Comment:
This duplicates the code in https://github.com/crowbar/crowbar-core/blob/204289d652f4a6c974c458595fdc8308c986b40f/chef/cookbooks/network/libraries/helpers.rb#L2. I would prefer if we had this code only once. @matthewoliver , when @guglo and I tried to use your code in `horizon_service.rb` we couldn't convince crowbar to use it. I am guessing it's a namespace issue or a missing include somewhere. Could you comment?

Reply:
I experimented more today, however I am newbee and not ruby savvy so I could not find a better way. Need help :)

Reply:
Looking closer at rails apps. We might be able to add something to `crowbar_framework/config/application.rb`. Like:

`config.eager_load_paths += <path into the chef/barclamp ruby code>`

Which will apparently extend the "scope" outside of the app/ directory. Once my current deploy is done I can have a play.

Reply:
Ok. So I've just been playing with hard coding paths.. because I want to know if it'll work.
I started with:
  `config.eager_load_paths += Dir["/opt/dell/chef"]`
But that died with not knowing a Node object was. There is a bunch of chef and other ruby code that may get in the way and cause issues. So instead I was more specific and tried:
`config.eager_load_paths += Dir["/opt/dell/chef/cookbooks/network/libraries"]`
I.e: http://pastebin.nue.suse.com/18645

And restarting crowbar (and crowbar-jobs) seemed to be fine with it. This includes attempting to use the NetworkHelper in `crowbar_framework/app/models/horizon_service.rb` with:
`node.crowbar["crowbar"]["links"]["OpenStack Dashboard (public)"] = "#{protocol}://#{NetworkHelper.wrap_ip(public_server_ip)}/"`

So from the surface it "looks" like it works.Looks kinda ugly having a static path. So it would be nice if we could find a way of making it more portable. 

But have run out of time to do any more playing today.

Reply:
@matthewoliver  Thanks for helping, tried config.eager_load_paths += Dir["#{config.root}/../chef/cookbooks/network/libraries"] 
it seems to be ok. Will update and remove the duplication.